### PR TITLE
7.x hooked access

### PR DESCRIPTION
**Pull request by adam-vessey**
Ontime #1733.
https://jira.duraspace.org/browse/ISLANDORA-885

### CODE REVIEW

File: includes/callback.inc [hunk 1]
```diff
@@ -35,7 +35,10 @@ function islandora_internet_archive_bookreader_dimensions_callback(AbstractObjec
  *   The page object whose OCR data-stream will be used for display.
  */
 function islandora_internet_archive_bookreader_text_callback(AbstractObject $object) {
-  echo isset($object['OCR']) ? htmlspecialchars($object['OCR']->content) : '';
+  $accessible_ocr = isset($object['OCR']) && islandora_datastream_access(FEDORA_VIEW_OBJECTS, $object['OCR']);
+  echo $accessible_ocr ?
+    htmlspecialchars($object['OCR']->content) :
+    t('Text not available!');
```
Comment (jonathangreen) [inline includes/callback.inc]: Should we maybe do a watchdog call here? With some debug info, so people can sort out why their OCR isn't working if this happens.

File: theme/theme.inc [hunk 2]
```diff
@@ -119,7 +119,7 @@ function theme_islandora_internet_archive_bookreader_book_info(array $variables)
  *   fields value(s).
  */
 function islandora_internet_archive_bookreader_info_fields(AbstractObject $object) {
-  if (!isset($object['MODS'])) {
+  if (!isset($object['MODS']) && islandora_datastream_access(FEDORA_VIEW_OBJECTS, $object['MODS'])) {
```
Comment (jonathangreen) [inline theme/theme.inc]: Am i not understanding the logic here? Should it be:
!islandora_datastream_access(FEDORA_VIEW_OBJECTS, $object['MODS'])

Comment (adam-vessey) [inline theme/theme.inc]: Good catch. Pretty sure I intended to do:
`!(isset($object['mods']) && islandora_datastream_access(FEDORA_VIEW_OBJECTS, $object['MODS']))`
(I've changed it to this)... Though on second thought:
`!isset($object['mods']) || !islandora_datastream_access(FEDORA_VIEW_OBJECTS, $object['MODS'])`
might read a little easier? Dunno.

File: includes/callback.inc (hunk 1 above)
Comment (adam-vessey) [inline includes/callback.inc]: Would breaking it up into the two separate conditions suffice, so we can report the different between `isset()` (the datastream doesn't exist) vs the `islandora_datastream_access()` (you don't have access to the datastream)?...  I find we occasionally get into polluting the watchdog logs.

Comment (jonathangreen) [inline includes/callback.inc]: Yah we could do that. I was just thinking if a client came in and was like: I get Text not available! its going to be a pain in the ass to debug why.

File: includes/callback.inc [hunk 3]
```diff
@@ -35,7 +35,14 @@ function islandora_internet_archive_bookreader_dimensions_callback(AbstractObjec
  *   The page object whose OCR data-stream will be used for display.
  */
 function islandora_internet_archive_bookreader_text_callback(AbstractObject $object) {
-  echo isset($object['OCR']) ? htmlspecialchars($object['OCR']->content) : '';
+  $ocr_exists = isset($object['OCR']);
+  $accessible_ocr = isset($object['OCR']) && islandora_datastream_access(FEDORA_VIEW_OBJECTS, $object['OCR']);
+  echo $ocr_exists? (
+    $accessible_ocr ?
+      htmlspecialchars($object['OCR']->content) :
+      t('You do not have permission to view the OCR for this page.')) :
```
Comment (jonathangreen) [inline includes/callback.inc:43]: We should make this call the permission error theme function you are creating @adam-vessey

**Pull request merged**